### PR TITLE
SAAS-175 Auth API

### PR DESCRIPTION
**Pull request by artemgavrilov**
- https://jira.percona.com/browse/SAAS-175

Implemented in: https://github.com/percona/pmm-managed/pull/420
Feature build: Percona-Lab/pmm-submodules#967

### CODE REVIEW

**Comment by AlekSi** [comment]
image

I really hope CI will catch that…

File: api/serverpb/server.proto [hunk 1]
```diff
@@ -218,4 +256,25 @@ service Server {
...
+  rpc SignIn(SignInRequest) returns (SignInResponse) {
+    option (google.api.http) = {
+      post: "/v1/Auth/SignIn"
+      body: "*"
+    };
+  }
+  // RefreshSession refreshes session timeout.
+  rpc RefreshSession(RefreshSessionRequest) returns (RefreshSessionResponse) {
+    option (google.api.http) = {
+      post: "/v1/Auth/RefreshSession"
```
Comment (AlekSi) [inline api/serverpb/server.proto:276]: There shouldn't be a need to expose that API – once user signed in, pmm-managed should refresh session in the background.

**Comment by AlekSi** [comment]
And yes, CI fails. Run `make init gen`

File: api/serverpb/server.proto (hunk 1 above)
Comment (artemgavrilov) [reply api/serverpb/server.proto]: Removed

**Comment by artemgavrilov** [comment]
@AlekSi sorry, fixed.

File: api/serverpb/server.proto [hunk 2]
```diff
@@ -218,4 +252,18 @@ service Server {
       body: "*"
     };
   }
+  // SignUp creates user Percona profile.
+  rpc SignUp(SignUpRequest) returns (SignUpResponse) {
+    option (google.api.http) = {
+      post: "/v1/Auth/SignUp"
```
Comment (AlekSi) [inline api/serverpb/server.proto:258]: Let me think about that URL…

**Comment by AlekSi** [comment]
> We need to add SaaS user email to Settings message

I will do that

**Comment by artemgavrilov** [comment]
Ok

File: api/serverpb/server.proto [hunk 3]
```diff
@@ -161,6 +163,32 @@ message AWSInstanceCheckRequest {
...
 
+message PlatformSignUpRequest {
+  // New Percona Platform user's email.
+  string email = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // New Percona Platform user's password.
+  string password = 2;
```
Comment (AlekSi) [inline api/serverpb/server.proto:174]: The presence of that field should be validated in code. In the future, we will relax that requirement. Removing a validator would be a breaking change from prototool perspective.

File: api/serverpb/server.proto [hunk 4]
```diff
@@ -161,6 +163,32 @@ message AWSInstanceCheckRequest {
...
+
+message PlatformSignInRequest {
+  // Existing Percona Platform user's email.
+  string email = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // Existing Percona Platform user's password.
+  string password = 2;
```
Comment (AlekSi) [inline api/serverpb/server.proto:187]: The presence of that field should be validated in code. In the future, we will relax that requirement. Removing a validator would be a breaking change from prototool perspective.

File: api/serverpb/server.proto [hunk 5]
```diff
@@ -218,4 +246,18 @@ service Server {
...
+      body: "*"
+    };
+  }
+  // PlatformSignIn links that PMM instance to Percona Platform user.
+  rpc PlatformSignIn(PlatformSignInRequest) returns (PlatformSignInResponse) {
+    option (google.api.http) = {
+      post: "/v1/Platform/SignIn"
+      body: "*"
+    };
+  }
```
Comment (AlekSi) [inline api/serverpb/server.proto:262]: Changed both URLs and method names

File: api/nginx/nginx.conf [hunk 6]
```diff
@@ -8,88 +11,13 @@ events {
 }
 
 http {
-  upstream managed-grpc {
```
Comment (BupycHuk) [inline api/nginx/nginx.conf:11]: Where it was moved?

Comment (AlekSi) [reply api/nginx/nginx.conf:11]: See the added comment on top of the file

**Pull request merged**